service/history/historyEngine.go

```diff
@@ -237,7 +237,7 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
 				// its length > 0 and has correct timestamp, to trkgger a db scan
 				fakeDecisionTask := []persistence.Task{&persistence.DecisionTask{}}
 				fakeDecisionTimeoutTask := []persistence.Task{&persistence.DecisionTimeoutTask{VisibilityTimestamp: now}}
-				e.txProcessor.NotifyNewTask(e.currentClusterName, now, fakeDecisionTask)
+				e.txProcessor.NotifyNewTask(e.currentClusterName, fakeDecisionTask)
```

Comment:
I think we should have the exact same contract on moving the clock on both timer and transfer processors.  Timer notification takes in the currentTime as notification while transfer does not.  I think we should unify these.

Reply:
transfer task do not have timestamp associated with it